crates/forge/tests/it/zk/nonce.rs

```diff
@@ -0,0 +1,48 @@
+use foundry_config::fs_permissions::PathPermission;
+use foundry_test_utils::{forgetest_async, util, util::OutputExt, TestProject};
+
+use crate::test_helpers::run_zk_script_test;
+
+forgetest_async!(setup_block_on_script_test, |prj, cmd| {
+    setup_deploy_prj(&mut prj);
+    run_zk_script_test(
+        prj.root(),
+        &mut cmd,
+        "./script/ScriptSetup.s.sol",
+        "ScriptSetupNonce",
+        None,
+        4,
+        Some(&["-vvvvv"]),
+    );
+});
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_zk_contract_nonce_mismatch() {
```

Comment:
why can't we use the usual test setup? because of the dep on `matter-labs/era-contracts`?
I actually don't see it i the test code (the dep), and even if we can hardcode the interfaces that we need (and the addresses we are talking to)